Tools/clinic/libclinic/codegen.py

```diff
@@ -188,3 +265,35 @@ def dump(self) -> str:
 
 
 DestinationDict = dict[str, Destination]
+
+
+class Codegen:
+    def __init__(self, limited_capi: bool) -> None:
+        self.limited_capi = limited_capi
+        self.ifndef_symbols: set[str] = set()
+        # dict: include name => Include instance
+        self.includes: dict[str, Include] = {}
+
+    def add_ifndef_symbol(self, name: str) -> bool:
+        if name in self.ifndef_symbols:
+            return False
+        self.ifndef_symbols.add(name)
+        return True
```

Comment:
I think this API is a little bit strange; would it be beneficial to split it in two APIs? One for adding the symbol and one for querying the list of symbols?